pkgs/common-updater/scripts/list-archive-versions

```diff
@@ -0,0 +1,50 @@
+#!/usr/bin/env bash
+
+# lists all available versions listed for a package in a site (http)
+
+archive=""  # archive url
+pname=""  # package name
+file=""  # file for writing debugging information
+
+while (( $# > 0 )); do
+    flag="$1"
+    shift 1
+    case "$flag" in
+        --url=*)
+            archive="${flag#*=}"
+            ;;
+        --pname=*)
+            pname="${flag#*=}"
+            ;;
+        --file=*)
+            file="${flag#*=}"
+            ;;
+        *)
+            echo "$0: unknown option ‘${flag}’"
+            exit 1
+            ;;
+    esac
+done
+
+# by default set url to the base dir of the first url in src.urls
+if [[ -z "$archive" ]]; then
+    archive="$(nix-instantiate $systemArg --eval -E \
+                   "with import ./. {}; dirOf (dirOf (lib.head $UPDATE_NIX_ATTR_PATH.src.urls))" \
+            | tr -d '"')"
+fi
+
+if [[ -z "$pname" ]]; then
+    pname="$UPDATE_NIX_ATTR_PATH"
+fi
+
+set +x
+
+# print a debugging message
+if [[ -n "$file" ]]; then
+    echo "# Listing versions for '$pname' at $archive" >> $file
+fi
+
+# print available versions
+tags=$(curl -sS "$archive/")
+tags=$(echo "$tags" | sed -rne "s,^.*<a href=\"$pname-([0-9.]+)\\.[^0-9].*\">.*,\\1,p")
```

Comment:
This appears to be overly restrictive, e.g. it skips pre-release versions like [enlightenment-0.18.0-rc1.tar.bz2](https://download.enlightenment.org/rel/apps/enlightenment/enlightenment-0.18.0-rc1.tar.bz2).

Reply:
We can try to provide a more general regex to match the version. But it may not be get one that matches most version formats. Any suggestion for a better regex?

Reply:
Done.

I have rewritten `list-archive-versions` in python, as it offers better regex handling. sed does not have non-greedy regex matching, which I needed.